File: src/frontend/src/types/sensor.ts
export interface SensorReading {
  cycle: number;
  s2?: number;  // HPC Outlet Temp (K)
  s3?: number;  // Combustor Outlet Temp (K)
  s4?: number;  // LPT Outlet Temp (K)
  s7?: number;  // HPC Pressure (psia)
  s8?: number;  // Fan Speed (rpm)
  s9?: number;  // Core Speed (rpm)
  s11?: number; // Static Pressure (psia)
  s12?: number; // Fuel Ratio (pps/psi)
  s14?: number; // Bypass Ratio
  s15?: number; // Bleed Enthalpy
  s17?: number; // HP Turbine Speed
  s20?: number; // HPT Coolant (lbf)
  s21?: number; // LPT Coolant (lbf)
  [key: string]: number | undefined;
}

export interface SensorMeta {
  key: string;
  name: string;
  unit: string;
  warningThreshold: number;
  criticalThreshold: number;
  normalRange: string;
}

File: src/backend/app/models/sensor.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey
from src.backend.app.database import Base

class SensorReading(Base):
    __tablename__ = "sensor_readings"

    id = Column(Integer, primary_key=True, index=True)
    asset_id = Column(String, ForeignKey("assets.asset_id"), index=True, nullable=False)
    cycle = Column(Integer, index=True, nullable=False)

    s2 = Column(Float, nullable=True)  # HPC Outlet Temp (K)
    s3 = Column(Float, nullable=True)  # Combustor Outlet Temp (K)
    s4 = Column(Float, nullable=True)  # LPT Outlet Temp (K)
    s7 = Column(Float, nullable=True)  # HPC Pressure (psia)
    s8 = Column(Float, nullable=True)  # Fan Speed (rpm)
    s9 = Column(Float, nullable=True)  # Core Speed (rpm)
    s11 = Column(Float, nullable=True) # Static Pressure (psia)
    s12 = Column(Float, nullable=True) # Fuel Ratio
    s14 = Column(Float, nullable=True) # Bypass Ratio
    s15 = Column(Float, nullable=True) # Bleed Enthalpy
    s17 = Column(Float, nullable=True) # HP Turbine Speed
    s20 = Column(Float, nullable=True) # HPT Coolant (lbf)
    s21 = Column(Float, nullable=True) # LPT Coolant (lbf)
